mesonbuild/mintro.py

```diff
@@ -317,15 +317,15 @@ def add_keys(opts: 'T.Union[dict[OptionKey, UserOption[Any]], cdata.KeyedOptionD
             elif isinstance(opt, options.UserComboOption):
                 optdict['choices'] = opt.printable_choices()
                 typestr = 'combo'
-            elif isinstance(opt, options.UserIntegerOption):
+            elif isinstance(opt, (options.UserIntegerOption, options.UserUmaskOption)):
                 typestr = 'integer'
             elif isinstance(opt, options.UserStringArrayOption):
                 typestr = 'array'
                 c = opt.printable_choices()
                 if c:
                     optdict['choices'] = c
             else:
-                raise RuntimeError("Unknown option type")
+                raise RuntimeError('Unknown option type: ', type(opt))
```

Comment:
Is it supported to pass a type as second argument of RuntimeError?

Reply:
Of course, something like a `match` statement that was required to be exhaustive would be even better